### PR TITLE
Transform vm_tests into isolated_tests

### LINKED ISSUE #222: Transform vm tests into isolated tests
Currently, we have 2 types of pure asm tests:
- `vm_tests`: older hand-written assembly tests that assume no initial state and check for reveal outputs
- `isolated_tests`: newer tests that use the `TestSpec` framework to pass in initial registers (rest are initialized as 0), initial fp, initial mem, and can check for post all.

We should transform all `vm_tests` into `isolated_tests` because the `TestSpec` framework is better, and we don't need to use `reveal` for these tests.

### PR DESCRIPTION
**Pull request by leoalt-ai**
Closes #222

## Motivation

The `vm_tests` module uses an older pattern (no initial state, check via `reveal` outputs). The `isolated_tests` module with the `TestSpec` framework is better: it supports initial register/RAM state, runs through 4 execution stages (raw, metered, preflight, mock proof), and tests at 3 different FP bases.

## Changes

- **Converted** all unique vm_tests to `TestSpec` format in `isolated_tests.rs`:
  - Additional mul tests (zero, one, powers of two, large numbers, max value, negative variants, commutative, chain, overflow)
  - Additional div tests (by one, equal, remainder, zero dividend, large, powers of two, chain, both negative, inverse relationship)
  - Additional comparison tests (lt_s positive/both_negative, gt_u false/equal, gt_s positive/both_negative, chain, edge cases, equivalence, mixed signed/unsigned)
  - Additional load/store roundtrip tests (storeb/storeh with offsets)
- **Moved** non-TestSpec tests (trap, input_hint, input_hint_with_frame_jump_and_xor) as standalone tests using `VmExecutor` directly
- **Dropped** redundant tests already covered by existing isolated_tests
- **Dropped** `test_basic_add_proof` (full proof generation is redundant with `mock_prove` constraint verification in TestSpec)
- **Deleted** `vm_tests.rs` module entirely

## Test plan

- [x] `cargo test --release -p womir-openvm-integration isolated_tests` — 165 tests pass
- [x] `cargo clippy --all-targets -- -D warnings` — clean
- [x] `cargo fmt --all -- --check` — clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

**Comment by leonardoalt** [comment]
I checked all tests that were moved/added and LGTM.

File: integration/src/tests/isolated_tests.rs [hunk 1]
```diff
@@ -2815,4 +2818,649 @@ mod tests {
...
+            ..Default::default()
+        };
+        test_spec_for_all_register_bases(spec)
+    }
+
+    #[test]
+    fn test_mul_large_numbers() {
+        setup_tracing_with_log_level(Level::WARN);
+
+        // 65537 * 65521 = 4294049777 (wrapping u32)
```
Comment (georgwiese) [inline integration/src/tests/isolated_tests.rs]:
```suggestion
        // 65537 * 65521 = 4294049777
```

It's not?

File: integration/src/tests/isolated_tests.rs [hunk 2]
```diff
@@ -2815,4 +2818,649 @@ mod tests {
...
+    #[test]
+    fn test_mul_negative_positive() {
+        setup_tracing_with_log_level(Level::WARN);
+
+        // (-5) * 3 = -15 = 0xFFFFFFF1
+        let spec = TestSpec {
+            program: vec![wom::mul(2, 0, 1)],
+            start_fp: 10,
+            start_registers: vec![(10, (-5_i32) as u32), (11, 3)],
+            expected_registers: vec![(12, 0xFFFFFFF1)],
```
Comment (georgwiese) [inline integration/src/tests/isolated_tests.rs]:
```suggestion
            expected_registers: vec![(12,(-15_i32) as u32)],
```

More readable IMO. Same applies to other tests.

File: integration/src/tests/isolated_tests.rs [hunk 3]
```diff
@@ -2815,4 +2818,649 @@ mod tests {
         };
         test_spec_for_all_register_bases(spec)
     }
+
+    // ==================== Additional Mul Tests (from vm_tests) ====================
```
[the remark this answers is not in the document]
Comment (leoalt-ai) [reply integration/src/tests/isolated_tests.rs]: Addressed: removed '(from vm_tests)' references, clustered similar tests, removed redundant tests. Also applied all remaining suggestions (readability, expected_ram, comparison test refactoring) in abf36cc.

File: integration/src/tests/isolated_tests.rs [hunk 4]
```diff
@@ -2815,4 +2818,649 @@ mod tests {
...
+            start_fp: 10,
+            start_registers: vec![(10, 999), (11, 1)],
+            expected_registers: vec![(12, 999)],
+            ..Default::default()
+        };
+        test_spec_for_all_register_bases(spec)
+    }
+
+    #[test]
+    fn test_mul_powers_of_two() {
```
[the remark this answers is not in the document]
Comment (leoalt-ai) [reply integration/src/tests/isolated_tests.rs]: Agreed — removed this test.
